app/models/work.rb

```diff
@@ -454,9 +454,13 @@ def challenge_claim_ids
 
   # Only allow a work to fulfill an assignment assigned to one of this work's authors
   def challenge_assignment_ids=(ids)
+    valid_ids = ids.reject(&:blank?)
+    valid_users = (self.users + [User.current_user]).compact
+
     self.challenge_assignments =
-      ids.map { |id| id.blank? ? nil : ChallengeAssignment.find(id) }.compact.
-      select { |assign| (self.users + [User.current_user]).compact.include?(assign.offering_user) }
+      ChallengeAssignment.where(id: valid_ids)
+        .compact
```

Comment:
`where` will never return an array containing `nil`, so calling `compact` here is unnecessary.